src/dolphin/downloadDolphin.ts

```diff
@@ -145,105 +146,110 @@ async function installDolphin(
-        log(`could not chmod/chown ${type} Dolphin\n${err}`);
+      case "linux": {
+        await installDolphinOnLinux(type, assetPath, dolphinPath, log);
+        break;
       }
-
-      // Move backed up User folder and user.json
-      if (alreadyInstalled) {
-        const oldUserFolder = path.join(backupLocation, "Slippi Dolphin.app", "Contents", "Resources", "User");
-        const newUserFolder = path.join(dolphinResourcesPath, "User");
-
-        if (await fs.pathExists(oldUserFolder)) {
-          log("moving User folder...");
-          await fs.move(oldUserFolder, newUserFolder);
-        } else {
-          log("no old User folder to move");
-        }
-        await fs.remove(backupLocation);
+      default: {
+        throw new Error(`Installing Netplay is not supported on this platform: ${process.platform}`);
       }
-      break;
     }
-    case "linux": {
-      // Delete the existing app image if there is one
-      try {
-        const dolphinAppImagePath = await findDolphinExecutable(type);
-        // Delete the existing app image if it already exists
-        if (await fileExists(dolphinAppImagePath)) {
-          log(`${dolphinAppImagePath} already exists. Deleting...`);
-          await fs.remove(dolphinAppImagePath);
-        }
-      } catch (err) {
-        log("No existing AppImage found");
-      }
+  } catch (err) {
+    log("Failed to install Dolphin...");
+    console.log(err);
+  }
+}
 
-      if (cleanInstall) {
-        const userFolder = type === DolphinLaunchType.NETPLAY ? "SlippiOnline" : "SlippiPlayback";
-        await fs.remove(path.join(app.getPath("home"), ".config", userFolder));
-      }
+async function installDolphinOnMac(assetPath: string, dolphinPath: string, log: (message: string) => void) {
+  const backupLocation = dolphinPath + "_old";
+  const dolphinResourcesPath = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "Resources");
 
-      const zip = new AdmZip(assetPath);
-      zip.extractAllTo(dolphinPath, true);
+  const alreadyInstalled = await fs.pathExists(dolphinResourcesPath);
+  let shouldRestore = false;
+  if (alreadyInstalled) {
+    log(`${dolphinResourcesPath} already exists. Moving...`);
+    await fs.move(dolphinPath, backupLocation);
+    shouldRestore = true;
+  }
 
-      // Make the file executable
-      const dolphinAppImagePath = await findDolphinExecutable(type);
-      log(`Setting executable permissions...`);
-      await fs.chmod(dolphinAppImagePath, "755");
-      break;
+  log(`Extracting to: ${dolphinPath}`);
+  await extractDmg(assetPath, dolphinPath);
+  await fs.readdir(dolphinPath, (err, files) => {
+    if (err) {
+      log(err.message);
+    }
+    if (files) {
+      files.forEach(async (file) => {
+        if (file !== "Slippi Dolphin.app") {
+          try {
+            await fs.unlink(path.join(dolphinPath, file));
+          } catch (err) {
+            fs.removeSync(path.join(dolphinPath, file));
+          }
+        }
+      });
```

Comment:
This seems overly complicated. Aren't we just trying to find and remove files which aren't `Slippi Dolphin.app`? Maybe we could do it a bit more functionally.

```suggestion
  const files = await fs.readdir(dolphinPath);
  await Promise.all(files.filter(file => file !== "Slippi Dolphin.app").map(async (file) => {
         try {
           await fs.unlink(path.join(dolphinPath, file));
          } catch (err) {
           await fs.remove(path.join(dolphinPath, file));
          }
      }));
```